main.py

```diff
@@ -19,67 +17,86 @@
 
 sniped = {}
 sniped_edit = {}
-
 #These dictionaries store the deleted messages that our on_message events capture
 
-
-
 class myBot(commands.Bot): #Being used as an event handler, but this class isn't needed for the bot to work overall (You'd just have to make the necessary changes). It's just more organized.
   async def on_ready(self):
       print(f"Logged in as {bot.user} (Version: {discord.__version__}") #When the bot is online this function prints.
 
-  async def on_message_delete(self, message): #When a message is deleted it looks at the channel that it happened in, the user who deleted it, (along with the other specifications) and appends it to the "sniped" dictionary.
-      sniped[message.channel.id] = [
-          message.content, message.author, message.attachments,
-          message.channel.name, message.created_at
-      ]
-
-  async def on_message_edit(self, before, after): #Same concept as on_message_delete
-      sniped_edit[before.channel.id] = [
-          before.content, before.author, before.channel.name,
-          before.created_at
-      ]
+  async def on_message_delete(self, message):
+        if message.author == self.user:
+            return # Ignore if the message was deleted by the bot itself
+        
+        sniped[message.channel.id] = [
+            message.content, message.author, message.attachments, message.stickers, message.channel.name, message.created_at
+        ]
 
+  async def on_message_edit(self, before, after):
+        sniped_edit[before.channel.id] = [
+           before.content, before.author, after.content, before.channel.name, before.created_at
+]
 
 bot = myBot(command_prefix=prefix, intents=intents)
 
-
-@bot.command()
-async def snipe(ctx):
+@bot.event
```

Comment:
remove on ready cmd (already defined in "myBot" class, also syncing in on_ready is bad practice cuz of ratelimits)

Reply:
Instead add

```
@bot.command()
@commands.is_owner()
async def sync_cmds(ctx):
    synced = await bot.tree.sync()
    print(f"Synced {len(synced)} command(s)")
```